core/trino-main/src/main/java/io/trino/execution/scheduler/EventDrivenFaultTolerantQueryScheduler.java

```diff
@@ -1525,6 +1530,12 @@ public void addPartition(int partitionId, NodeRequirements nodeRequirements)
             DataSize defaultTaskMemory = stage.getFragment().getPartitioning().equals(COORDINATOR_DISTRIBUTION) ?
                     getFaultTolerantExecutionDefaultCoordinatorTaskMemory(session) :
                     getFaultTolerantExecutionDefaultTaskMemory(session);
+            if (activeNodesCount == 1 && stage.getFragment().getRoot().getSources().stream()
+                    .anyMatch(planNode -> planNode instanceof RefreshMaterializedViewNode)) {
+                // REFRESH MATERIALIZED VIEW will issue other SQL commands under the hood. If its task memory is
+                // non-zero, then a deadlock scenario is possible if we only have a single node in the cluster.
+                defaultTaskMemory = DataSize.ofBytes(0);
+            }
```

Comment:
I wonder if we really need to check `activeNodesCount`. As `REFRESH MATERIALIZED VIEW` itself is just a HTTP call under the hood.

Reply:
It may be or it may be not. In Trino itself we only have Iceberg MVs, which are implemented without external calls and don't exhibit this issue.

Reply:
But I agree that we probably can set this unconditionally.